ClientTests/TabSessionTests.swift

```diff
@@ -0,0 +1,527 @@
+        tabManager.resetProcessPool()
+        tabManager.reset()
+        
+        
+        let group = DispatchGroup()
+        
+        group.enter()
+        History.deleteAll({
+            group.leave()
+        })
+        
+        group.enter()
+        HTTPCookieStorage.shared.removeCookies(since: .distantPast)
+        WKWebsiteDataStore.default().httpCookieStore.getAllCookies({
+            $0.forEach({
+                group.enter()
+                WKWebsiteDataStore.default().httpCookieStore.delete($0, completionHandler: {
+                    group.leave()
+                })
+            })
+            group.leave()
+        })
+        
+        group.enter()
+        WKWebsiteDataStore.default().removeData(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(),
+                                                modifiedSince: .distantPast,
+                                                completionHandler: {
+            group.leave()
+        })
+
+        group.notify(queue: .main, execute: {
+            completion()
+        })
+    }
+    
+    func testPrivateTabSessionSharing() {
+        let dataStoreExpectation = XCTestExpectation(description: "dataStorePersistence")
+        let cookieStoreExpectation = XCTestExpectation(description: "cookieStorePersistence")
+        var webViewNavigationAdapter = WebViewNavigationAdapter()
+        
+        destroyData {
+            let urls = ["https://bing.com",
+                        "https://google.com",
+                        "https://yandex.ru",
+                        "https://yahoo.com"]
+            
+            self.tabManager.addTabsForURLs(urls.compactMap({ URL(string: $0) }), zombie: false, isPrivate: true)
+            XCTAssertTrue(self.tabManager.allTabs.count == 4, "Error: Not all Tabs are created equally")
+            
+            let group = DispatchGroup()
+            self.tabManager.allTabs.forEach({
+                XCTAssertNotNil($0.webView, "WebView is not created yet")
+                if $0.webView?.configuration.websiteDataStore.isPersistent == true {
+                    XCTFail("Private Tab is storing data persistently!")
+                }
+                
+                group.enter()
+            })
+            
+            webViewNavigationAdapter = WebViewNavigationAdapter(didFailListener: { _ in
+                group.leave()
+            }, didFinishListener: {
+                group.leave()
+            })
+            self.tabManager.addNavigationDelegate(webViewNavigationAdapter)
+            
+            group.notify(queue: .main) {
+                // All requests finished loading.. time to check the cookies..
+                let group = DispatchGroup()
+                self.tabManager.allTabs.forEach({
+                    group.enter()
+                    $0.webView?.configuration.websiteDataStore.fetchDataRecords(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(), completionHandler: { records in
```

Comment:
I am not sure, but will this not also check if the cookies are there or not. If so isn' the check below
`$0.webView?.configuration.websiteDataStore.httpCookieStore.getAllCookies(` redundant?

Reply:
Yes, but I'm not only checking for cookies.. I'm checking for disk cache, mem cache, etc.
The separate call for just cookies alone works too. I don't know if the websiteDataStore records and cookieStore are linked. That's why I did both just in case.

Are you sure they're linked or?

Reply:
I think so, can you please verify :)

Reply:
Fixed. You are correct.